src/Verify.MSTest.Tests/Tests.cs

```diff
@@ -1,14 +1,29 @@
-﻿// ReSharper disable UnusedParameter.Local
+﻿// TODO: Add a test that uses the base class to prevent regressions until it's deleted
+// TODO: Add a test for when the [TestClass] already has a TestContext property
+
+#region WillBeSourceGenerated
+partial class Tests
+{
+    [SuppressMessage("Performance", "CA1822:Mark members as static", Justification = "Injected by the test framework per test.")]
+    public TestContext TestContext
```

Comment:
@SimonCropp the downside of the MSTest extensibility model is that we need to create this property in the _user's_ class, so there could be compilation errors if they already have the property defined. I think this is ok because `VerifyBase` also defined this property, so people moving from the base class already couldn't define this property themselves.

However, if users also wanted to hook get or set like we are, they couldn't with my simple code. I _could_ create a private setter method to update the async local, and if there's a conflict document that users should call our helper, however I don't like that because:

1. I don't think it's a very common scenario
2. That would then require a private property to store the TestContext, and add yet another field to the user's test class, which feels dirtier.

For reference that hook-able generated code would look something like this

```csharp
partial class Tests
{
    private TestContext? __verifyTestContext;

    public TestContext TestContext
    {
        get => __verifyTestContext;
        set
        {
            __verifyTestContext = value;
            OnSetTestContext(value);
        }

        private static void OnSetTestContext(TestContext context)
        {
            CurrentTestContext.Value = context;
        }
}
```

Reply:
>  I think this is ok because VerifyBase also defined this property, so people moving from the base class already couldn't define this property themselves.

but can they use the generated one? ie in the base class case, if they want to leverage TestContext they could use the one from the base.

Reply:
what if we name the generated property something that wont conflict eg `public TestContext VerifyTestContext`

Reply:
I can't name the generated property something different, as MSTest is doing property injection by name.

To start I'm only generating the property if one doesn't already exist and relying on the exception message in `Verifier` to help correct any issues a user might have.

Also note that this would be limited to users that need to hook the _set_. If they only want to get the TestContext they can call our generated property.

If you disagree with that plan let me know and I can use the above technique to give users a hook.